app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsViewModel.kt

```diff
@@ -499,6 +522,41 @@ class NotificationsViewModel @Inject constructor(
             }
     }
 
+    fun hasNewestNotificationId(type: Notification.Type, statusId: String, notificationId: String): Boolean {
+        val trackerArray = when(type) {
+            Notification.Type.FAVOURITE -> seenFavorites
+            Notification.Type.REBLOG -> seenBoosts
+            else -> null
+        } ?: return true
+
+        val highestNotificationId = trackerArray[statusId]
+
+        return if (highestNotificationId == null || isEqualOrNewer(notificationId, highestNotificationId)) {
+            trackerArray[statusId] = notificationId
+
+            true
+        } else {
+            // TODO edge case: a newer favorite has been removed: the old notification will not be added again
+            //    (because the removed id is still in the seen array)
+            //    The code could find this out only heuristically: "looking at these notification ids (range), one in the array is not amongst them"
+
+            false
+        }
+    }
+
+    /**
+     * NOTE this currently assumes that the ids are integers. Can that change? If it does all notifications are unequal.
+     */
+    fun isEqualOrNewer(thisId: String, idToCompare: String): Boolean {
+        try {
+            return thisId.toInt() >= idToCompare.toInt()
```

Comment:
There are non-int ids out there...

(From the chat about problems with Hometown notifications:

`01GZ6K8XSQ6RMPXTFJPZRCXSXJ
01GZ6JS7V2K93VB54S312BC2WN
01GZ6HWAQSQMTCVTT2RXJC0ENE
01GZ6HVX4N5VE2MVATZHGXGRFJ
01GZ6HJQT3DSE358C2BQ0WN6KK`

This would still work with the frequent "length and lexicographical" comparissons I've seen in the code...

Reply:
`StringUtils.kt` already has `String.isLessThan`, which would the right thing to use here.

Reply:
Ok, done.